packages/EDA/src/package.ts

```diff
@@ -628,3 +556,58 @@ export function isApplicableLinearRegression(df: DG.DataFrame, predictColumn: DG
 
   return true;
 }
+
+//name: trainPLSRegression
+//meta.mlname: PLS Regression
+//meta.mlrole: train
+//input: dataframe df
+//input: column predictColumn
+//input: int components = 3 {min: 1; max: 10} [Number of latent components]
+//output: dynamic model
+export async function trainPLSRegression(df: DG.DataFrame, predictColumn: DG.Column, components: number): Promise<Uint8Array> {
+  const features = df.columns;
+
+  if (components > features.length)
+    throw new Error('Number of components is greater than features count');
+
+  const model = new PlsModel();
+  await model.fit(features, predictColumn, components);
+
+  return model.toBytes();
+}
+
+//name: applyLinearRegression
+//meta.mlname: PLS Regression
+//meta.mlrole: apply
+//input: dataframe df
+//input: dynamic model
+//output: dataframe table
+export function applyPLSRegression(df: DG.DataFrame, model: any): DG.DataFrame {
+  const unpackedModel = new PlsModel(model);
+  return DG.DataFrame.fromColumns([unpackedModel.predict(df.columns)]);
+}
+
+//name: isApplicablePLSRegression
+//meta.mlname: PLS Regression
+//meta.mlrole: isApplicable
+//input: dataframe df
+//input: column predictColumn
+//output: bool result
+export function isApplicablePLSRegression(df: DG.DataFrame, predictColumn: DG.Column): boolean {
```

Comment:
Can you also add isInteractive? Format is the same and you can find examples in the master

Reply:
Added

Reply:
Concerning the previous questions...
The main computations are performed at the wasm side (we also use them in pls-ml):
1) input: features (X) and target (Y)
2) output: regression coeffs, T,U-scores, P,Q-loadings (see https://en.wikipedia.org/wiki/Partial_least_squares_regression for more details), prediction
An important feature is that X, Y are normalized. So, regression coeffs DO NOT contain the bias term. In pls-ml, we need this term. For this reason, we perform another step - we compute it at the TS side.

Concerning loadings
For visualization purposes, we need loadings and explained variances. And they should be packed with regression coefs. For convenience of packing, we append each expl.variance to the corresponding loadings array.

So, in pls-ml, we use almost the same as in mulitvariate analysis in combination with some specific extras

Reply:
ok, LGTM